src/routes/shoppingCart.js

```diff
@@ -1,11 +1,13 @@
 import express from 'express';
-import { listItems, addItem,  updateItem, removeItem } from "../controllers";
+import { listItems, addItem,  updateItem, removeItem } from '../controllers';
+import { validateRequest } from '../middleware';
+import { addItemSchema, removeItemSchema, updateItemSchema } from '../validation';
 
 const shoppingCartRouter = express.Router();
 
 shoppingCartRouter.get('/', listItems);
-shoppingCartRouter.post('/', addItem);
-shoppingCartRouter.patch('/', updateItem);
-shoppingCartRouter.delete('/', removeItem);
+shoppingCartRouter.post('/', validateRequest(addItemSchema, 'body'), addItem);
+shoppingCartRouter.patch('/', validateRequest(updateItemSchema, 'body'), updateItem);
+shoppingCartRouter.delete('/', validateRequest(removeItemSchema, 'body'), removeItem);
```

Comment:
Será q nao era legal fazer cada middleware numa linha, pra nao ficar uma linha muito longa?
```suggestion
shoppingCartRouter.delete('/',
  validateRequest(removeItemSchema, 'body'),
  removeItem
);
```